provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/aws.rb

```
@@ -0,0 +1,267 @@
+      @result['result']['ssh-auth']['identityfile'] = @aws_keyfile unless @aws_keyfile.nil?
+      @result['status'] = 0
+    rescue Exception => e
+      log.error('Unexpected Error Occurred in FogProviderAWS.create:' + e.inspect)
+      @result['stderr'] = "Unexpected Error Occurred in FogProviderAWS.create: #{e.inspect}"
+    else
+      log.debug "Create finished successfully: #{@result}"
+    ensure
+      @result['status'] = 1 if @result['status'].nil? || (@result['status'].is_a?(Hash) && @result['status'].empty?)
+    end
+  end
+
+  def confirm(inputmap)
+    providerid = inputmap['providerid']
+    fields = inputmap['fields']
+    begin
+      # Our fields are fog symbols
+      fields.each do |k,v|
+        instance_variable_set('@' + k, v)
+      end
+      # Run EC2 credential validation
+      validate!
+      # Confirm server
+      log.debug "Invoking server confirm for id: #{providerid}"
+      server = self.connection.servers.get(providerid)
+      # Wait until the server is ready
+      raise 'Server #{server.id} is in ERROR state' if server.state == 'ERROR'
+      log.debug "waiting for server to come up: #{providerid}"
+      server.wait_for(600) { ready? }
+
+      # Handle tags
+      hashed_tags = {}
+      @tags.map{ |t| key,val=t.split('='); hashed_tags[key]=val} unless @tags.nil?
+      # Always set the Name tag, so we display correctly in AWS console UI
+      unless hashed_tags.keys.include?('Name')
+        hashed_tags['Name'] = @task['config']['hostname']
+      end
+      create_tags(hashed_tags, providerid) unless hashed_tags.empty?
+
+      bootstrap_ip =
+        if server.public_ip_address
+          server.public_ip_address
+        else
+          Resolv.getaddress(server.dns_name) unless server.dns_name.nil?
+        end
+      if bootstrap_ip.nil?
+        log.error 'No IP address available for bootstrapping.'
+      else
+        log.debug "Bootstrap IP address #{bootstrap_ip}"
```

we should raise an exception here